### ISSUE TITLE
Patches to typescript files .ts not affecting transpiled JS files in <module>/dist

### ISSUE THREAD

**Issue by larsblumberg** [ds300/patch-package#242]
I've patched a typescript (`.ts`) file in a module in `node_modules/<module>/src` and integrated the patch toolchain with package.json as documented.

The patches are applied properly (even when reinstalling the module) but their transpiled `.js` files in `<module>/dist/` aren't affected, hence the monkey patch has no affect in my application.

What am I missing?

**Comment by papb** [comment]
You're missing that packages written in typescript are transpiled by the repository owners upon publishing, so you changing the `ts` source should not have any effect. In fact ideally the module should not even have published the `ts` files to begin with.

**Comment by larsblumberg** [comment]
Ah, I didn’t know that. As the `/dist` wasn’t included with the module’s report, I assumed that the .js files get generated on my side in the moment I install the module.

**Comment by papb** [comment]
@larsblumberg In theory, both cases are possible:

* The `.js` files are generated in your side;
* The `.js` files are generated at the moment of release in the library owner's side, and you just get them directly

However, I do not remember ever seeing the first case, and would not recommend this approach. It could be done via a `postinstall` npm hook in the module, but for this to work npm would have to first install the typescript compiler (and worse, other typescript-only utilities they might use would have to be installed as well) in order to be able to build it for you. This would make installation much slower, since npm would end up installing all that packages `devDependencies` first.

The second case is the normal, and what I would recommend. As a module creator, one can use whichever tools they like for development, such as typescript, coffeescript, babel, whatever. But this should not be a problem of the end user, so you should just publish the final generated `.js` files in your module (and ideally not even publish the source `.ts` files along, since they will be just useless files increasing the installation size).

However, if you wish to say what is the specific package, I can help you check if by any chance it uses the first approach.

Since the first approach is rare and not recommended, I am not sure `patch-package` intends to support this use case. If it does, you might have indeed found a bug here.

**Comment by larsblumberg** [comment]
@papb Thanks for your elaborated answer. I have figured that the library I wanted to patch - https://github.com/auth0/auth0-spa-js - uses the regular, second approach. Since its generated JS code is minified, I refrained from patching it.

I guess with the ever increasing usage of TypeScript in the JS ecosystem there will be less use cases for this patching library, although it does work really great. I used it only the minified JS files to see how it works, but then that's not a sustainable approach because the resulting diff isn't of any value since the minified JS file is a one-liner.
